packages/at_secondary_server/lib/src/verb/handler/notify_verb_handler.dart

```diff
@@ -71,7 +71,8 @@ class NotifyVerbHandler extends AbstractVerbHandler {
       var currentAtSign = AtSecondaryServerImpl.getInstance().currentAtSign;
       // If '@' is missing before an atSign, the formatAtSign method prefixes '@' before atSign.
       verbParams[FOR_AT_SIGN] = AtUtils.formatAtSign(verbParams[FOR_AT_SIGN]);
-      verbParams[AT_SIGN] = AtUtils.formatAtSign(verbParams[AT_SIGN]);
+      verbParams[AT_SIGN] =
+          AtUtils.formatAtSign(verbParams[AT_SIGN]) ?? currentAtSign;
```

Comment:
This doesn't look right. Why are you setting `verbParams[AT_SIGN]` to currentAtSign if `verbParams[AT_SIGN]` wasn't supplied?

Reply:
When messageType is "text", by syntax, sharedBy atSign is not populated. So, false will be returned when validating if atSign is authorized to send notifications (null not equal to currentAtSign). Hence verbParam[AT_SIGN] is set to currentAtSign. 
When messageType is "text", verbParams[AT_SIGN] is not added to the notification key.

Reply:
But at this point you haven't checked if you are in an authenticated, polAuthenticated or un-authenticated

Reply:
Is it ok to move the logic to set currentAtSign to verbParams[AT_SIGN] into "_handleAuthenticatedConnection" method?

Reply:
Yes I think so.

Just make sure your overall logic works for auth or polAuth connections, and maybe throw an exception for unauthenticated connections

Reply:
Sure Gary.